documentation/src/docs/controller-service.md

```diff
@@ -47,403 +47,362 @@ You may obtain a copy of the License at
+clients to query for the next Segments, the stream store exposes via the
+Controller Service efficient queries for finding immediate successors
+and predecessors for any arbitrary Segment.  
+
+To enable serving queries like those mentioned above, we need to efficiently store a time series of these Segment transitions and index them against time. We store this information about the current and historical state of a Stream Segments in a set of records which are designed to optimize on aforementioned queries. Apart from Segment-specific metadata record, the current state of Stream comprises of other metadata types that are described henceforth.
+
+#### Records  
+
+Stream time series is stored as a series of records where each record corresponds to an epoch. As Stream scales and transitions from one epoch to another, a new record is created that has complete information about Stream Segments that forms the epoch.
+
+- **Epoch Records:**  
+_Epoch: ⟨time, list-of-segments-in-epoch⟩_.
+We store the series of _active_ Stream Segments as they transition from one epoch to another into individual epoch records. Each epoch record corresponds to an epoch which captures a logically consistent (as defined earlier) set of Stream Segments that form the Stream and are valid through the lifespan of the epoch. The epoch record is stored against the epoch number. This record is optimized to answer to query Segments from an epoch with a single call into the store that also enables retrieval of all Stream Segment records in the epoch in _O(1)_. This record is also used for fetching a Segment-specific record by first computing Stream Segment's creation epoch from Stream Segment ID and then retrieving the epoch record.
+
+ - **Current Epoch:**
+ A special epoch record called `currentEpoch`. This is the currently _active_ epoch in the Stream. At any time exactly one epoch is marked as the current epoch. Typically this is the latest epoch with the highest epoch number. However, during an ongoing Stream update workflow like _scale_ or _rolling Transaction_, the current epoch may not necessarily be the latest epoch. However, at the completion of these workflows, the current epoch is marked as the latest epoch in the stream.
+
+  The following are three most commonly used scenarios where we want to efficiently know the set of Segments that form the Stream:
+   1. _Initial set of Stream Segments_: The **head** of the Stream computation is very efficient as it is typically either the first epoch record or the latest truncation record.
+   2. _Current set of Stream Segments_: The **tail** of the Stream is identified by the current epoch record.
+   3. _Successors of a particular Stream Segment_: The successor query results in two calls into the store to retrieve Stream Segment's sealed epoch and the corresponding epoch record. The successors are computed as the Stream Segments that overlap with the given Stream Segment.
+
+- **Segment Records:**
```

Comment:
Why this is not a bullet as the previous ones?

Reply:
Accepted suggestion